### PR TITLE
feat(product): inject extra product driver response transforms

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?
Adds a token for providing extra transform behavior to the Magento product driver.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
This is in preparation for #1642

### CODE REVIEW

**Review by lderrickable — CHANGES_REQUESTED** [review]

File: libs/product/driver/magento/src/injection-tokens/transforms/response.token.spec.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    });
+
+    magentoProductFactory = TestBed.inject(MagentoProductFactory);
+    responseTransform = TestBed.inject(MAGENTO_PRODUCT_RESPONSE_TRANSFORM);
+
+    magentoProduct = magentoProductFactory.create();
+    result = responseTransform(magentoProduct, 'test');
+  });
+
+  it('should run the standard transform first, followed by the injected transforms', () => {
```
[the remark this answers is not in the document]
Comment (griest024) [reply libs/product/driver/magento/src/injection-tokens/transforms/response.token.spec.ts:48]: if it hadn't run first, the `id` would be something like `'undefined transform 1 transform 2'`. The fact that `response.id` is truthy in the first transform is what ensures that it ran first.

File: libs/product/driver/magento/src/injection-tokens/transforms/response.token.ts [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+import { DaffMagentoProductResponseTransform } from '../../interfaces/public_api';
+import { MagentoProduct } from '../../models/magento-product';
+import { transformMagentoProductResponse } from '../../transforms/public_api';
+import { DAFF_PRODUCT_MAGENTO_EXTRA_PRODUCT_RESPONSE_TRANSFORMS } from './response-extra.token';
+
+/**
+ * An internal token to combine the Magento reponse transform with the injected transforms.
+ */
+export const MAGENTO_PRODUCT_RESPONSE_TRANSFORM = new InjectionToken<DaffMagentoProductResponseTransform>(
```
[the remark this answers is not in the document]
Comment (griest024) [reply libs/product/driver/magento/src/injection-tokens/transforms/response.token.ts]: yeah changed

File: libs/product/driver/magento/src/injection-tokens/transforms/response.token.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+import { DaffProduct } from '@daffodil/product';
+import { DaffProductDriverResponse } from '@daffodil/product/driver';
+
+import { DaffMagentoProductResponseTransform } from '../../interfaces/public_api';
+import { MagentoProduct } from '../../models/magento-product';
+import { transformMagentoProductResponse } from '../../transforms/public_api';
+import { DAFF_PRODUCT_MAGENTO_EXTRA_PRODUCT_RESPONSE_TRANSFORMS } from './response-extra.token';
+
+/**
+ * An internal token to combine the Magento reponse transform with the injected transforms.
```
[the remark this answers is not in the document]
Comment (griest024) [reply libs/product/driver/magento/src/injection-tokens/transforms/response.token.ts]: fixed

File: libs/product/driver/magento/src/injection-tokens/transforms/response-extra.token.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+} from '@angular/core';
+
+// workaround https://github.com/graycoreio/daffodil/issues/1667
+import { DaffProduct } from '@daffodil/product';
+import { DaffProductDriverResponse } from '@daffodil/product/driver';
+
+import { DaffMagentoProductResponseExtraTransform } from '../../interfaces/public_api';
+
+/**
+ * An multi-provider injection token for providing extra transform logic in the Product Magento driver.
```
[the remark this answers is not in the document]
Comment (griest024) [reply libs/product/driver/magento/src/injection-tokens/transforms/response-extra.token.ts]: fixed

**Review by lderrickable — APPROVED** [review]

**Pull request merged**